src/Chapter2.hs

```diff
@@ -501,7 +528,8 @@ True
 >>> isThird42 [42, 42, 0, 42]
 False
 -}
-isThird42 = error "isThird42: Not implemented!"
+isThird42 :: (Eq a, Num a) => [a] -> Bool
+isThird42 xs = length xs >= 3 && xs !! 2 == 42
```

Comment:
Unfortunately, the current solution doesn't work on infinite lists. Moreover, its performance is not good because `length` for lists is slow.

In this task, you need to use explicit pattern matching on lists to match on the third element.